MPCONFIG properties can not be used inside @DataSourceDefinition /CUSTCOM-196

# Description #
----------

Payara Micro is not able to use MicroProfile Config properties inside ``@DataSourceDefinition`` annotation.

```java
// hardcoded version works
@DataSourceDefinition(
    name            = "java:global/jdbc/reproducer",
    className       = "org.postgresql.ds.PGSimpleDataSource", 
    serverName      = "localhost",
    portNumber      = 5432,
    databaseName    = "postgres",
    user            = "postgres",
    password        = "mysecretpassword"
)

// fully microprofile config does not work
@DataSourceDefinition(
    name            = "java:global/jdbc/reproducer",
    className       = "${MPCONFIG=dataSource.className}", 
    serverName      = "${MPCONFIG=dataSource.serverName}",
    portNumber      = 5432,
    databaseName    = "${MPCONFIG=dataSource.databaseName}",
    user            = "${MPCONFIG=dataSource.user}",
    password        = "${MPCONFIG=dataSource.password}"
)

// partial microprofile config does not work
@DataSourceDefinition(
    name            = "java:global/jdbc/reproducer",
    className       = "org.postgresql.ds.PGSimpleDataSource", 
    serverName      = "${MPCONFIG=dataSource.serverName}",
    portNumber      = 5432,
    databaseName    = "${MPCONFIG=dataSource.databaseName}",
    user            = "${MPCONFIG=dataSource.user}",
    password        = "${MPCONFIG=dataSource.password}"
)
```

## Expected Outcome

Ability to configure my DataSource using MicroProfile config.

## Current Outcome

Exception is thrown, due to the MPCONFIG properties does not get replaced.

```bash
[2020-02-13T11:35:45.809+0100] [] [SEVERE] [] [javax.enterprise.system.core] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1581590145809] [levelValue: 1000] [[
Exception while preparing the app : Exception [EclipseLink-4002] (Eclipse Persistence Services - 2.7.4.payara-p2): org.eclipse.persistence.exceptions.DatabaseException
Internal Exception: java.sql.SQLException: Error in allocating a connection. Cause: Class name is wrong or classpath is not set for : ${MPCONFIG=dataSource.className}
Error Code: 0
```

or

```bash
Caused by: java.net.UnknownHostException: ${MPCONFIG=dataSource.serverName}
```

## Steps to reproduce (Only for bug reports) 

See reproducer at https://github.com/haraldatbmw/payara-micro-mpconfig

## Context (Optional)

I want to use payara micro and have the ability to fully configure my application using MicroProfile Config. I want to use H2 in-memory db at dev time and Postgres as Integration and Production environment.

## Environment ##

- **Payara Version**: 5.194
- **Edition**: Micro
- **JDK Version**: 8 OpenJDK 
- **Operating System**: Windows
- **Database**: PostGres

--- a/appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/common/DataSourceSpec.java
+++ b/appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/common/DataSourceSpec.java
@@ -37,7 +37,7 @@
  * only if the new code is made subject to such option by the copyright
  * holder.
  */
-// Portions Copyright [2016-2019] [Payara Foundation and/or its affiliates]
+// Portions Copyright [2016-2020] [Payara Foundation and/or its affiliates]
 
 package com.sun.gjc.common;
 
@@ -46,6 +46,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import org.glassfish.config.support.TranslatedConfigView;
 
 /**
  * Encapsulate the DataSource object details obtained from
@@ -124,7 +125,7 @@ public class DataSourceSpec implements java.io.Serializable {
      * @param value    Value of property to be set.
      */
     public void setDetail(int property, String value) {
-        details.put(property, value);
+        details.put(property, TranslatedConfigView.expandConfigValue(value));
     }
 
     /**

--- a/nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/ApplicationLifecycle.java
+++ b/nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/ApplicationLifecycle.java
@@ -487,7 +487,8 @@ public class ApplicationLifecycle implements Deployment, PostConstruct {
             events.send(new Event<>(Deployment.DEPLOYMENT_BEFORE_CLASSLOADER_CREATION, context), false);
 
             context.createApplicationClassLoader(clh, handler);
-
+            tempAppInfo.setAppClassLoader(context.getFinalClassLoader());
+            
             events.send(new Event<>(Deployment.AFTER_APPLICATION_CLASSLOADER_CREATION, context), false);
 
             // this is a first time deployment as opposed as load following an unload event,
@@ -1809,11 +1810,7 @@ public class ApplicationLifecycle implements Deployment, PostConstruct {
             return false;
         }
 
-        if (virtEnv != null && virtEnv.isPaasEnabled()) {
-            return true;
-        }
-
-        return false;
+        return virtEnv != null && virtEnv.isPaasEnabled();
     }
 
     // gets the default target when no target is specified for non-paas case
@@ -1838,7 +1835,7 @@ public class ApplicationLifecycle implements Deployment, PostConstruct {
            // for other cases, we try to derive it from domain.xml
            List<String> targets =
                domain.getAllReferencedTargetsForApplication(appName);
-           if (targets.size() == 0) {
+           if (targets.isEmpty()) {
                throw new IllegalArgumentException("Application not registered");
            }
            if (targets.size() > 1) {
@@ -2461,7 +2458,7 @@ public class ApplicationLifecycle implements Deployment, PostConstruct {
         if (appInfo.getSniffers().size() > 0) {
             for (Sniffer sniffer : appInfo.getSniffers()) {
                 if (sniffer.isUserVisible()) {
-                    sb.append(sniffer.getModuleType() + ", ");
+                    sb.append(sniffer.getModuleType()).append(", ");
                 }
             }
         }

